### ISSUE TITLE
iquest (4.1.9) How to get RESC_LOC for a file in a compound resource?

### ISSUE THREAD

**Issue by keithj** [irods/irods#3288]
For a non-compound resource we can perform a query like this:

```
iquest "select RESC_NAME, RESC_LOC where COLL_NAME = '/seq/test' and DATA_NAME = 'test.txt'"
RESC_NAME = testResc
RESC_LOC = iceland
------------------------------------------------------------
```

The result shows the host `iceland` as `RESC_LOC`.

For a compound resource e.g.

```
root:passthru
└── replicate:replication
    ├── seq-green:random
    │   ├── green1:random
    │   │   ├── seq-r1:unix file system
    │   │   ├── seq-r2:unix file system
    │   │   ├── seq-r3:unix file system

etc.
```

A similar query for any file that has been stored in the compound resource always
reports the root resource as the location, rather than the leaf resource where the
data are located.

e.g.

```
iquest "select RESC_NAME, RESC_LOC where COLL_NAME  = '/seq/test2' and DATA_NAME = 'test.txt'"
RESC_NAME = root
RESC_LOC = EMPTY_RESC_HOST
------------------------------------------------------------
```

I'm interested in the host where `test.txt` is located. Is there a way to determine this with iquest?

Thanks.

**Comment by jasoncoposky** [comment]
select DATA_RESC_HIER which maps to the COL_D_RESC_HIER column:
https://github.com/irods/irods/blob/4-1-stable/iRODS/lib/core/include/rodsGenQueryNames.h#L76

**Comment by keithj** [comment]
I tried that query during my experiment, but the host is not part of the hierarchy string:
e.g.

`DATA_RESC_HIER = passthru_0;roundrobin_0;unixfs_8`

It appears that I would need to add parsing step to extract the leaf and then another iquest query to
look up the host in `r_resc_main`. Is that correct?

**Comment by jasoncoposky** [comment]
that is correct for 4.1.9.  we have fixed this in 4.2 with the move to using leaf resource ids

**Comment by trel** [comment]
This works as expected in 4.2.x.   Closing.
